pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -24,132 +24,153 @@ import (
+			log.Error(err, err.Error(), "namespace", req.Namespace, "name", certificateRequest.Spec.IssuerRef.Name)
 			return ctrl.Result{}, err
 		}
 		googleCASClient, ok = casClient.(cas.Client)
 		if !ok {
 			err := errors.New("Retrieved a non-CAS client from the cache, this should never happen")
-			log.Error(err, err.Error(), "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Couldn't retrieve CAS Issuer client %s: %v", issuerNamespaceName, errors.New("couldn't retrieve CAS Issuer client"))
+			log.Error(err, err.Error(), "namespace", req.Namespace, "name", certificateRequest.Spec.IssuerRef.Name)
 			return ctrl.Result{}, err
 		}
 	case "GoogleCASIssuer":
 		issuer := api.GoogleCASIssuer{}
 		issuerNamespaceName := types.NamespacedName{
 			Namespace: req.Namespace,
-			Name:      cr.Spec.IssuerRef.Name,
+			Name:      certificateRequest.Spec.IssuerRef.Name,
 		}
 		if err := r.Client.Get(ctx, issuerNamespaceName, &issuer); err != nil {
-			log.Error(err, "failed to retrieve GoogleCASIssuer resource", "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Failed to retrieve CAS Issuer resource %s: %v", issuerNamespaceName, err)
 			return ctrl.Result{}, err
 		}
 		parent = fmt.Sprintf("projects/%s/locations/%s/certificateAuthorities/%s", issuer.Spec.Project, issuer.Spec.Location, issuer.Spec.CertificateAuthorityID)
 		casClient, ok := issuerctrl.Issuers.Load(issuerNamespaceName)
 		if !ok {
 			err := errors.New("couldn't retrieve CAS Issuer client - it probably hasn't reconciled yet")
-			log.Error(err, err.Error(), "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Couldn't retrieve CAS Issuer client %s: %v", issuerNamespaceName, errors.New("couldn't retrieve CAS Issuer client"))
+			log.Error(err, err.Error(), "namespace", req.Namespace, "name", certificateRequest.Spec.IssuerRef.Name)
 			return ctrl.Result{}, err
 		}
 		googleCASClient, ok = casClient.(cas.Client)
 		if !ok {
 			err := errors.New("Retrieved a non-CAS client from the cache, this should never happen")
-			log.Error(err, err.Error(), "namespace", req.Namespace, "name", cr.Spec.IssuerRef.Name)
-			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "Couldn't retrieve CAS Issuer client %s: %v", issuerNamespaceName, errors.New("couldn't retrieve CAS Issuer client"))
+			log.Error(err, err.Error(), "namespace", req.Namespace, "name", certificateRequest.Spec.IssuerRef.Name)
 			return ctrl.Result{}, err
 		}
 	default:
-		log.Info("Noticed unhandled kind in Certificate Request:", "kind", cr.Spec.IssuerRef.Kind)
+		log.Info("Noticed unhandled kind in Certificate Request:", "kind", certificateRequest.Spec.IssuerRef.Kind)
```

Comment:
Maybe log this as an error and set an error condition on the CR? Up to you.

Reply:
👍 @maelvls also pointed this out whenever I logged an error